aibolit/metrics/cognitiveC/cognitive_c.py

```diff
@@ -2,36 +2,64 @@
 from aibolit.patterns.var_middle.var_middle import JavalangImproved
 from typing import List
 
+increment_for = [
+    javalang.tree.IfStatement,
+    javalang.tree.SwitchStatement,
+    javalang.tree.ForStatement,
+    javalang.tree.WhileStatement,
+    javalang.tree.DoStatement,
+    javalang.tree.CatchClause,
+    javalang.tree.BreakStatement,
+    javalang.tree.ContinueStatement,
+    javalang.tree.TernaryExpression,
+    # javalang.tree.BinaryOperation,
+]
+
+nested_for = [
+    javalang.tree.IfStatement,
+    javalang.tree.SwitchStatement,
+    javalang.tree.ForStatement,
+    javalang.tree.WhileStatement,
+    javalang.tree.DoStatement,
+    javalang.tree.CatchClause,
+]
+
 
 class CognitiveComplexity:
     '''
     beta version of Cognitive Complexity
     '''
     def __init__(self):
-        pass
+        self.complexity = 0
 
-    def if_statements(self, nodes: List) -> int:
-        count = 0
-        for i in nodes:
-            if type(i.node) in [javalang.tree.IfStatement]:
-                count += 1
-        return count
+    def get_complexity(self, childs, nested_level=0):
+        child_arr = childs if isinstance(childs, List) else [childs]
 
-    def for_statements(self, nodes: List) -> int:
-        count = 0
-        for i in nodes:
-            if type(i.node) in [javalang.tree.ForStatement]:
-                count += 1
-        return count
+        for c in child_arr:
+
+            if hasattr(c, 'children'):
+                if type(c) in increment_for and type(c) in nested_for:
+                    self.complexity += (1 + nested_level)
+                    for each_child in c.children:
+                        self.get_complexity(each_child, nested_level + 1)
+
+                elif type(c) in increment_for and type(c) not in nested_for:
+                    self.complexity += 1
+                    for each_child in c.children:
+                        self.get_complexity(each_child, nested_level)
+                else:
+                    for each_child in c.children:
```

Comment:
```
for each_child in c.children:
   self.get_complexity(each_child, nested_level)
```

this code used three time, please optimize

Reply:
fixed